openeogeotrellis/geopysparkdatacube.py

```diff
@@ -362,9 +362,22 @@ def _normalize_temporal_reducer(self, dimension: str, reducer: str) -> str:
 
     @callsite
     def add_dimension(self, name: str, label: str, type: str = None):
+        cube = self
+        if type== "temporal":
+            if cube.metadata.has_temporal_dimension():
+                raise OpenEOApiException(status_code=400, code="DimensionAlreadyExist", message=
```

Comment:
Better: log an error message if this happens, then follow up in kibana logs if this occurs in practice.
Then a next step can be to convert the error log in an exception? 
(Unless the current version would also result in an exception already.)